File: backend/app/schemas/equipment.py
# Schémas pour les équipements
from pydantic import BaseModel, validator
from typing import Optional, Dict, Any
from datetime import datetime, date
from enum import Enum

class EquipmentStatus(str, Enum):
    ACTIVE = "active"
    INACTIVE = "inactive"
    MAINTENANCE = "maintenance"
    BROKEN = "broken"

class EquipmentCriticality(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"

class EquipmentBase(BaseModel):
    name: str
    model: Optional[str] = None
    serial_number: Optional[str] = None
    manufacturer: Optional[str] = None
    purchase_date: Optional[date] = None
    installation_date: Optional[date] = None
    warranty_expiry: Optional[date] = None
    expected_lifespan: Optional[int] = None
    site_id: Optional[int] = None
    production_line_id: Optional[int] = None
    status: EquipmentStatus = EquipmentStatus.ACTIVE
    criticality: EquipmentCriticality = EquipmentCriticality.MEDIUM
    specifications: Optional[Dict[str, Any]] = None

class EquipmentCreate(EquipmentBase):
    @validator('serial_number')
    def validate_serial_number(cls, v):
        if v is not None and len(v.strip()) == 0:
            return None
        return v

class EquipmentUpdate(BaseModel):
    name: Optional[str] = None
    model: Optional[str] = None
    serial_number: Optional[str] = None
    manufacturer: Optional[str] = None
    purchase_date: Optional[date] = None
    installation_date: Optional[date] = None
    warranty_expiry: Optional[date] = None
    expected_lifespan: Optional[int] = None
    site_id: Optional[int] = None
    production_line_id: Optional[int] = None
    status: Optional[EquipmentStatus] = None
    criticality: Optional[EquipmentCriticality] = None
    specifications: Optional[Dict[str, Any]] = None

class EquipmentResponse(EquipmentBase):
    id: int
    created_at: datetime
    updated_at: datetime
    status_display: Optional[str] = None
    criticality_display: Optional[str] = None
    
    class Config:
        from_attributes = True

# Schéma avec les relations pour les détails complets
class EquipmentWithRelations(EquipmentResponse):
    site_name: Optional[str] = None
    production_line_name: Optional[str] = None

    class Config:
        from_attributes = True

# Schéma pour les filtres de recherche
class EquipmentFilter(BaseModel):
    site_id: Optional[int] = None
    production_line_id: Optional[int] = None
    status: Optional[EquipmentStatus] = None
    criticality: Optional[EquipmentCriticality] = None
    search: Optional[str] = None  # Recherche dans nom, modèle, fabricant

File: frontend/src/types/equipment.ts
// Types pour les équipements
export interface Site {
  id: number;
  name: string;
  location?: string;
  description?: string;
  created_at: string;
  updated_at: string;
}

export interface ProductionLine {
  id: number;
  site_id: number;
  name: string;
  description?: string;
  created_at: string;
  updated_at: string;
}

export type EquipmentStatus = 'active' | 'inactive' | 'maintenance' | 'broken';
export type EquipmentCriticality = 'low' | 'medium' | 'high' | 'critical';

export interface Equipment {
  id: number;
  name: string;
  model?: string;
  serial_number?: string;
  manufacturer?: string;
  purchase_date?: string;
  installation_date?: string;
  warranty_expiry?: string;
  expected_lifespan?: number;
  site_id?: number;
  production_line_id?: number;
  status: EquipmentStatus;
  criticality: EquipmentCriticality;
  specifications?: Record<string, any>;
  created_at: string;
  updated_at: string;
  status_display?: string;
  criticality_display?: string;
}

export interface EquipmentWithRelations extends Equipment {
  site_name?: string;
  production_line_name?: string;
}

export interface EquipmentFilter {
  site_id?: number;
  production_line_id?: number;
  status?: EquipmentStatus;
  criticality?: EquipmentCriticality;
  search?: string;
}

export interface EquipmentCreate {
  name: string;
  model?: string;
  serial_number?: string;
  manufacturer?: string;
  purchase_date?: string;
  installation_date?: string;
  warranty_expiry?: string;
  expected_lifespan?: number;
  site_id?: number;
  production_line_id?: number;
  status?: EquipmentStatus;
  criticality?: EquipmentCriticality;
  specifications?: Record<string, any>;
}

export interface EquipmentUpdate {
  name?: string;
  model?: string;
  serial_number?: string;
  manufacturer?: string;
  purchase_date?: string;
  installation_date?: string;
  warranty_expiry?: string;
  expected_lifespan?: number;
  site_id?: number;
  production_line_id?: number;
  status?: EquipmentStatus;
  criticality?: EquipmentCriticality;
  specifications?: Record<string, any>;
}

export interface EquipmentStats {
  total: number;
  by_status: {
    active: number;
    maintenance: number;
    broken: number;
    inactive: number;
  };
  by_criticality: {
    critical: number;
    high: number;
    medium: number;
    low: number;
  };
}

// Constantes pour les options
export const EQUIPMENT_STATUS_OPTIONS = [
  { value: 'active', label: 'Actif', color: 'green' },
  { value: 'inactive', label: 'Inactif', color: 'gray' },
  { value: 'maintenance', label: 'En maintenance', color: 'orange' },
  { value: 'broken', label: 'En panne', color: 'red' }
] as const;

export const EQUIPMENT_CRITICALITY_OPTIONS = [
  { value: 'low', label: 'Faible', color: 'green' },
  { value: 'medium', label: 'Moyenne', color: 'yellow' },
  { value: 'high', label: 'Élevée', color: 'orange' },
  { value: 'critical', label: 'Critique', color: 'red' }
] as const;
